### ISSUE TITLE
add possibility to enforce rules no-unused-*

### ISSUE THREAD

**Issue by max-norin** [future-architect/eslint-plugin-vue-scoped-css#49]
Now `no-unused-keyframes` and `no-unused-selector` rules only work if scoped style. There are situations when without scoped style I would like to use this rules. But there is no such possibility.
It might be worth adding a new option `scoped: Boolean`

**Comment by ota-meshi** [comment]
Thank you for this issue.

I think if you don't use `<style scoped>` you will get a lot of false positives.
Why do you still want to use it?

**Comment by max-norin** [comment]
If you use the scoped style, then the css file size is increased by adding the `.selector[data-v- {hash}]` attributes.
Our team uses unique component names and class names of the same name for them. 
If the vue component does not contain other vue components, then you can not use the scoped style. In this case, all styles must be wrapped in the class of the component's top tag.
For example,
```vue
// DefaultHeader.vue
<template>
  <header class="mn-default-header">
    <nav class="mn-banner">
      <a class="flex-center" href="/" role="banner"> banner </a>
    </nav>
    <nav class="mn-nav">
      <a href="/1"> 1 </a>
      <a href="/2"> 2 </a>
    </nav>
  </header>
</template>

<script lang="ts">
import { Component, Vue } from 'nuxt-property-decorator'

@Component({})
export default class DefaultHeader extends Vue {}
</script>

<style lang="postcss">
.mn-default-header {
  z-index: 20;

  .mn-banner {
    z-index: 1;
    a {
      z-index: 3;
    }
  }
  .mn-nav {
    z-index: 2;
    a {
      z-index: 4;
    }
  }
}
</style>
```
```vue
// DefaultFooter.vue
<template>
  <header class="mn-default-footer">
    <nav class="mn-banner">
      <a class="flex-center" href="/" role="banner"> banner </a>
    </nav>
    <nav class="mn-nav">
      <a href="/1"> 1 </a>
      <a href="/2"> 2 </a>
    </nav>
  </header>
</template>

<script lang="ts">
import { Component, Vue } from 'nuxt-property-decorator'

@Component({})
export default class DefaultFooter extends Vue {}
</script>

<style lang="postcss">
.mn-default-footer {
  z-index: 20;

  .mn-banner {
    z-index: 1;
    a {
      z-index: 3;
    }
  }
  .mn-nav {
    z-index: 2;
    a {
      z-index: 4;
    }
  }
}
</style>
```
In this case, the styles will not overlap.
